optaplanner-core/src/main/java/org/optaplanner/core/impl/heuristic/selector/entity/nearby/NearEntityNearbyEntitySelector.java

```diff
@@ -42,7 +43,12 @@
     public NearEntityNearbyEntitySelector(EntitySelector childEntitySelector, EntitySelector originEntitySelector,
             NearbyDistanceMeter nearbyDistanceMeter, NearbyRandom nearbyRandom, boolean randomSelection) {
         this.childEntitySelector = childEntitySelector;
-        this.originEntitySelector = originEntitySelector;
+        if (!(originEntitySelector instanceof MimicReplayingEntitySelector)) {
+            // In order to select a nearby entity, we must first have something to be near by.
+            throw new IllegalStateException("Programming error: Nearby entity selector (" + this +
```

Comment:
"Programming error" isn't a term we use often. It's not an "impossible state", I guess?